src/main/java/com/liveramp/daemon_lib/utils/JobletProcessHandler.java

```diff
@@ -41,8 +41,14 @@ public void onRemove(ProcessDefinition<M, Pid> watchedProcess) throws DaemonExce
     }
 
     if (jobletStatusManager.exists(identifier)) {
+      JobletStatus status;
+      try {
+        status = jobletStatusManager.getStatus(identifier);
+      } catch (Exception e) {
+        status = JobletStatus.IN_PROGRESS;
```

Comment:
I do not have full context on the goals here, but I am highly skeptical of eaten exceptions -- can we LOG.error the exception, even if we aren't re-throwing it?

Reply:
Depending on what the parent code does, it might even make sense to catch this exception, run the failure callback, and then re-throw the exception.  In in case you want to mark the handler as "broken" (so it doesn't keep getting work to do and failing it).

Reply:
Basically agreeing with Ben here, I think the correct way to do this is to have the `catch` block try to run the failure callback, rather than messing with control flow in this way. You can, inside the catch block, wrap the failure callback call and throw some explicit exception type for when it's the failure callback itself that's doing the failing.

Reply:
i definitely hear you on the flow control. one of the reasons i was a little reticent to do this, is that if i just 1) catch the exception, 2) attempt to run failure callback 3) re-throw same exception or a new one then 
```
        jobletStatusManager.remove(identifier);
        configStorage.deleteConfig(identifier);
```
are _not_ going to get called, so there's some additional clean up that's not going to happen. that being said, that's no worse than what currently happens in this case, so i think you guys are right. i'll make this change.

Reply:
i just pushed the change. lmk if this is addresses your concerns.